app/cafe.py

```diff
@@ -0,0 +1,19 @@
+import datetime
+
+from app.errors import (NotVaccinatedError,
+                        OutdatedVaccineError,
+                        NotWearingMaskError)
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, visitor: dict) -> str:
+        if not visitor.get("vaccine"):
+            raise NotVaccinatedError("All friends should be vaccinated")
+        if visitor["vaccine"].get("expiration_date") < datetime.date.today():
+            raise OutdatedVaccineError("Outdated vaccine")

```

Comment:
The error message here could be more descriptive. Instead of 'Outdated vaccine', consider something like 'The visitor's vaccine is outdated. Visitors must have a valid, non-expired vaccine to enter the cafe.' This gives a clearer explanation of what the problem is.